mlflow/utils/download_cloud_file_chunk.py

```diff
@@ -41,20 +41,11 @@ def main():
             download_path=args.download_path,
             http_uri=args.http_uri,
         )
-    except (ConnectionError, ChunkedEncodingError):
+    except Exception:
         with open(args.temp_file, "w") as f:
+            # Retry all download failures at least once
             json.dump({"retryable": True}, f)
         raise
-    except HTTPError as e:
-        with open(args.temp_file, "w") as f:
-            json.dump(
-                {
-                    "retryable": e.response.status_code in (401, 403, 408),
```

Comment:
@BenWilson2 @harupy There are many other status codes where we probably want at least one retry. See https://github.com/mlflow/mlflow/blob/4e947e943908217e292fb7cc686cf88fc1aa675f/mlflow/utils/request_utils.py#L17-L26. How did we decide to special case these ones?

Reply:
IIRC our thinking was that 5xx errors from cloud providers generally weren't immediately recoverable. That was an assumption and wasn't based on any in-depth investigation, though. 
Upon looking at this again, it makes sense to at least give any failure another shot.